cql3/statements/modification_statement.cc

```diff
@@ -275,13 +276,36 @@ modification_statement::do_execute(query_processor& qp, service::query_state& qs
         return execute_with_condition(qp, qs, options);
     }
 
-    return execute_without_condition(qp, qs, options).then([] (coordinator_result<> res) {
+    return execute_without_condition(qp, qs, options).then([this, &options, &qp] (coordinator_result<> res) {
         if (!res) {
             return make_ready_future<::shared_ptr<cql_transport::messages::result_message>>(
                     seastar::make_shared<cql_transport::messages::result_message::exception>(std::move(res).assume_error()));
         }
-        return make_ready_future<::shared_ptr<cql_transport::messages::result_message>>(
+
+        json_cache_opt json_cache = maybe_prepare_json_cache(options);
+        std::vector<dht::partition_range> keys = build_partition_keys(options, json_cache);
+    
+        if (keys.size() != 1) {
+            return make_ready_future<::shared_ptr<cql_transport::messages::result_message>>(
```

Comment:
Batch statements can have many keys, we should handle them.

Reply:
by not attaching tablet info if there are many keys.

Reply:
Yes, that's why I attach tablet info only if there is one element in `keys`, is something wrong in that approach?